adal/src/telemetry/java/com/microsoft/aad/adal/HttpEvent.java

```diff
@@ -97,47 +97,34 @@ void setXMsCliTelemData(final CliTelemInfo cliTelemInfo) {
             return;
         }
 
-        if (!StringExtensions.isNullOrBlank(cliTelemInfo.getServerErrorCode())
-                && !cliTelemInfo.getServerErrorCode().equals("0")) {
-            setServerErrorCode(cliTelemInfo.getServerErrorCode());
-        }
-
-        if (!StringExtensions.isNullOrBlank(cliTelemInfo.getServerSubErrorCode())
-                && !cliTelemInfo.getServerSubErrorCode().equals("0")) {
-            setServerSubErrorCode(cliTelemInfo.getServerSubErrorCode());
-        }
-
-        if (!StringExtensions.isNullOrBlank(cliTelemInfo.getRefreshTokenAge())) {
-            setRefreshTokenAge(cliTelemInfo.getRefreshTokenAge());
-        }
-
-        if (!StringExtensions.isNullOrBlank(cliTelemInfo.getSpeRing())) {
-            setSpeRing(cliTelemInfo.getSpeRing());
-        }
+        setServerErrorCode(cliTelemInfo.getServerErrorCode());
```

Comment:
Will there be a chance that the ServerErrorCode is null? What will happen then?

Reply:
The `error_code` should be returned by the service even if no error is encountered (`0` indicates no error) -- that said, if it _doesn't_ come back it'll be caught by the check inside of `*Event` classes that only sets the `error_code` and `suberror_code` is `!StringExtensions.isNullOrBlank()` returns `false`